### DISCUSSION TITLE
Have Renovate use a dependency update type as commit type

### DISCUSSION THREAD

**Discussion by Mjb141** [renovatebot/renovate#31312] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab

### Please tell us more about your question or problem

I would like Renovate to use the underlying dependency update type as the commit type, rather than defaulting to `chore` for all dependency updates.

As an example:

Say our image is tagged `2.4.0` and the first line is `FROM image:1.2`, and `image` releases a major version upgrade from `1.2` to `2.0`

Renovate currently opens an MR `chore(deps): update image 1.2 to 2.0` but we want the type of update (in this case major) to be the type of increment of our image, so Renovate would instead open `major(deps): update image from 1.2 to 2.0` _for that specific dependency_ only, and our next version would be `3.0.0`.

Is this possible?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by RahulGautamSingh** [comment]
You can define package rules for this. Use `matchUpdateTypes` and `semanticCommitType`.

Eg:

```json
{
 "packageRules": [
  {
   "matchUpdateTypes": ["major"],
   "semanticCommitType": "major"
  }
 ]
}
```
